MonoGame.Framework/Graphics/GraphicsDevice.cs

```diff
@@ -1278,7 +1278,10 @@ public void DrawUserPrimitives<T>(PrimitiveType primitiveType, T[] vertexData, i
                 throw new ArgumentOutOfRangeException("primitiveCount");
 
             if (vertexDeclaration == null)
-                throw new ArgumentNullException("vertexDeclaration");
+                throw new ArgumentNullException("vertexDeclaration");
+
+            if (vertexDeclaration.VertexStride < ReflectionHelpers.FastSizeOf<T>())
+                throw new ArgumentOutOfRangeException("vertexDeclaration", "Vertex stride of vertexDeclaration should be at least as big as the stride of the actual vertices.");

```

Comment:
Is this really true?

Like i could see someone passing a `MyStruct[]` that happens to include multiple verts in it.

Is the more correct check is if the size of the `vertexData` is a multiple of the `vertexDeclaration.VertexStride` ?

@mrhelmut @harry-cpp ?

Reply:
I might be unaware of something, but there are 3 methods in this class that take `VertexDeclaration` as an argument. And 2 of them does this exact check. One of them doesn't - the subject one.

Here:
https://github.com/MonoGame/MonoGame/blob/ba5c722a399fa4e192859a7dd17377b20ae4d327/MonoGame.Framework/Graphics/GraphicsDevice.cs#L1237-L1238

And here:
https://github.com/MonoGame/MonoGame/blob/ba5c722a399fa4e192859a7dd17377b20ae4d327/MonoGame.Framework/Graphics/GraphicsDevice.cs#L1157-L1158